RS/2D vector space/main.py

```diff
@@ -0,0 +1,137 @@
+import pygame, sys, time, random, math
+from pygame.locals import *
+
+pygame.init()
+pygame.mixer.init()
+
+# Declare all variables etc
+BLACK = (0, 0, 0)
+WHITE = (255, 255, 255)
+RED = (255, 0, 0)
+GREEN = (0, 255, 0)
+BLUE = (0, 0, 255)
+
+windowWidth = 256
+windowHeight = 256
+
+tailLength = 20
+lumTotal = 0
+lumCount = 0
+doingSomethingToPassTheTime = 0
+firstPart = True
+secondPart = False
+
+window = pygame.display.set_mode((windowWidth, windowHeight))
+picture = pygame.image.load('pic1.jpg')
+picture = pygame.transform.scale(picture, (windowWidth, windowHeight))
+
+window.blit(picture, (0, 0))
+
+
+while firstPart == True:
+
+    for event in pygame.event.get():
+        if event.type == QUIT:
+            pygame.quit()
+            sys.exit()
+
+    pxArray = pygame.PixelArray(picture)
+
+    for x in range(0, windowWidth - 1):
+        for y in range(0, windowHeight - 1):
+            currentRed = window.get_at((x, y)).r
+            currentGreen = window.get_at((x, y)).g
+            currentBlue = window.get_at((x, y)).b
+            currentAlpha = window.get_at((x, y)).a
+
+            pxArray[currentRed, currentGreen] = RED
+            pxArray[currentRed, currentBlue] = BLACK
+            pxArray[currentGreen, currentRed] = GREEN
+            pxArray[currentGreen, currentBlue] = BLACK
+            pxArray[currentBlue, currentRed] = BLUE
+            pxArray[currentBlue, currentGreen] = BLACK
+
+    print ('checkpoint2')
+
+    pict = pxArray.make_surface()
+
+    del pxArray
+
+    print('checkpoint3')
+
+    window.blit(pict, (0, 0))
+
+    pygame.display.update()
+
+    print('checkpoint4')
+
+    firstPart = False
+    secondPart = True
```

Comment:
The while loops are redundant with the boolean variables switching like this. Again --- could have simply incremented an integer, or refactored completely into separate functions.